README.md

```diff
@@ -1,3 +1,151 @@
+│       ├── seeders     # Model factories
+│       ├── utils       # Utilities
+│       ├── validation  # TypeBox validation utilities
+│       ├── app.ts      # Firebase app initialization
+│       └── index.ts    # Cloud functions entrypoint
+│
+└── webapp     # Webapp frontend source code
+    ├── node_modules
+    └── src               # Frontend source code
+        ├── app           # Angular JS 17 app & components
+        ├── assets        # Static assets
+        ├── environments  # Environment configuration
+        └── lib           # API, types & utilities
+```
+
+## Firebase
+
+This project uses Firebase as its backend. The configuration can be found in the [`.firebaserc`](./.firebaserc) and [`firebase.json`](./firebase.json) files.
+
+Install the firebase cli tools with the following commands:
+
+```bash
+npm install -g firebase-tools
+firebase login  # Login to the Google account destined to manage the Firebase project
+```
+
+### Emulators
+
+In order to run offline and for development purposes, we use Java Firebase emulators.
+
+```bash
+export JAVA_TOOL_OPTIONS="-Xmx4g"  # Set your desired max RAM (here: 4GB)
+firebase emulators:start  # Start the emulators
+```
+
+You will then be able to access the following UIs:
+
+- Authentication UI: `http://localhost:4000/auth`
+- Emulator UI: `http://localhost:4000`
+- Firestore UI: `http://localhost:4000/firestore`
+- Cloud Functions UI: `http://localhost:4000/functions`
+
+### Firestore
+
+We use Firestore as our database. The rules are located in the [`firestore.rules`](./firestore.rules) file.
+Basically, the database cannot be externally accessed, and must be interacted with through cloud functions.
+
+A database prototype schema can be found [here on dbdiagrams.io](https://dbdiagram.io/d/Firebase-LLM-Mediation-660d473a03593b6b61123f24) (readonly, change it with your own if you take-on the project).
+
+You can seed the database with the default data by running the following command:
+
+```bash
+curl "http://127.0.0.1:5001/friendlychat-d6dc5/us-central1/seedDatabase?seeder_password=seeder_password"
+```
+
+This calls a cloud functions that adds to the database the default data. You may want to clear it first.
+
+### Cloud Functions
+
+We use Firebase Cloud Functions to run server-side code. The functions are located in the [`functions`](./functions) directory.
+
+```bash
+cd functions
+cp .env.example .env  # Copy the example environment file
+npm install  # Install the dependencies
+npm run build:watch # Build the functions and watch for file changes for rebuilding
+```
+
+Upon running `npm run build`, if the emulator is running, it will automatically reload the functions.
+
+### Authentication
+
+This project sets up Firebase Authentication with email/password and Google sign-in.
+
+#### Email - Password authentication
+
+By default, experiment participants log in using their `uid`. It is passed to the Firebase Authentication service as `email:{uid}@palabrate` and `password:{uid}`.
```

Comment:
I think we'd be better to avoid participants having to have a user account on firebase. If they have that, then we need to do a lot more user-management work. Users being completely anonymous will be much better for us.

Reply:
(so then we'd say that: only admins/experimenters have an account on firebase)

Reply:
Currently, a participant only sees the uid they are assigned, after the experimenter generates the uids and shares with the usparticipant. So technically, users never interact with their accounts besides logging in with their uid (the dummy password is handled by the frontend app). 

Otherwise, we may have trouble if users accidentally close their browser for a short period of time (by the way we should discuss the expected behavior when users abandon an experiment).

Reply:
We should check with Crystal, but I think they get sent their UID by prolific/the expérimenter, so we should not need to link their email to it on firebase... 

I think the key security consideration and assumption here is that participant uniqueness is defined by unique URL (basically we use a security model for participants like photos links). 

This is my best estimate of the right balance on balancing pragmatics, privacy, and security. I'll also make sure to do a review with security/privacy experts too.

Reply:
Fixed in #8 - 94e43b38773459f1d1e078ff98a42155c22840d6
Participants will "log in" using their UID, without Firebase Authentication